scaleway/table_scaleway_kubernetes_node.go

```diff
@@ -0,0 +1,277 @@
+	region := plugin.GetMatrixItem(ctx)["region"].(string)
+
+	parseRegionData, err := scw.ParseRegion(region)
+	if err != nil {
+		plugin.Logger(ctx).Error("scaleway_kubernetes_node.listKubernetesNodes", "region_parsing_error", err)
+		return nil, err
+	}
+
+	quals := d.KeyColumnQuals
+	if quals["region"] != nil && quals["region"].GetStringValue() != region {
+		return nil, nil
+	}
+
+	// Create client
+	client, err := getSessionConfig(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("scaleway_kubernetes_node.listKubernetesNodes", "connection_error", err)
+		return nil, err
+	}
+
+	// Get cluster details
+	clusterData := h.Item.(*k8s.Cluster)
+
+	// Create SDK objects for Scaleway Kubernetes product
+	kubernetesApi := k8s.NewAPI(client)
+
+	req := &k8s.ListNodesRequest{
+		Region:    parseRegionData,
+		ClusterID: clusterData.ID,
+		Page:      scw.Int32Ptr(1),
+	}
+	// Additional filters
+	if quals["name"] != nil {
+		req.Name = scw.StringPtr(quals["name"].GetStringValue())
+	}
+
+	// Retrieve the list of nodes
+	maxResult := int64(100)
+
+	// Reduce the basic request limit down if the user has only requested a small number of rows
+	limit := d.QueryContext.Limit
+	if d.QueryContext.Limit != nil {
+		if *limit < maxResult {
+			maxResult = *limit
+		}
+	}
+	req.PageSize = scw.Uint32Ptr(uint32(maxResult))
+
+	var count int
+
+	for {
+		resp, err := kubernetesApi.ListNodes(req)
+		if err != nil {
+			plugin.Logger(ctx).Error("scaleway_kubernetes_node.listKubernetesNodes", "query_error", err)
+			return nil, err
+		}
+
+		for _, node := range resp.Nodes {
+			d.StreamListItem(ctx, node)
+
+			// Increase the resource count by 1
+			count++
+
+			// Context can be cancelled due to manual cancellation or the limit has been hit
+			if d.QueryStatus.RowsRemaining(ctx) == 0 {
+				return nil, nil
+			}
+		}
+
+		if resp.TotalCount == uint32(count) {
+			break
+		}
+		req.Page = scw.Int32Ptr(*req.Page + 1)
+	}
+
+	return nil, nil
+}
+
+//// HYDRATE FUNCTIONS
+
+func getKubernetesNode(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	region := plugin.GetMatrixItem(ctx)["region"].(string)
+
+	parseRegionData, err := scw.ParseRegion(region)
+	if err != nil {
+		plugin.Logger(ctx).Error("scaleway_kubernetes_node.getKubernetesNode", "region_parsing_error", err)
+		return nil, err
+	}
+
+	if d.KeyColumnQuals["region"].GetStringValue() != region {
+		return nil, nil
+	}
+
+	// Create client
+	client, err := getSessionConfig(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("scaleway_kubernetes_node.getKubernetesNode", "connection_error", err)
+		return nil, err
+	}
+
+	// Create SDK objects for Scaleway Kubernetes product
+	kubernetesApi := k8s.NewAPI(client)
+
+	id := d.KeyColumnQuals["id"].GetStringValue()
+	KubernetesRegion := d.KeyColumnQuals["region"].GetStringValue()
+
+	// No inputs
+	if id == "" && KubernetesRegion == "" {
+		return nil, nil
+	}
```

Comment:
```suggestion
	id := d.KeyColumnQuals["id"].GetStringValue()

	// No inputs
	if id == "" {
		return nil, nil
	}
```

I don't think we need to add an additional check, since empty region check is already handled in - 
```
	if d.KeyColumnQuals["region"].GetStringValue() != region {
		return nil, nil
	}
```